crates/cheatcodes/src/inspector.rs

```diff
@@ -293,18 +293,14 @@ pub struct Cheatcodes {
     /// All recorded ETH `deal`s.
     pub eth_deals: Vec<DealRecord>,
 
-    /// Holds the stored gas info for when we pause gas metering. It is an `Option<Option<..>>`
-    /// because the `call` callback in an `Inspector` doesn't get access to
-    /// the `revm::Interpreter` which holds the `revm::Gas` struct that
-    /// we need to copy. So we convert it to a `Some(None)` in `apply_cheatcode`, and once we have
-    /// the interpreter, we copy the gas struct. Then each time there is an execution of an
-    /// operation, we reset the gas.
-    pub gas_metering: Option<Option<Gas>>,
-
-    /// Holds stored gas info for when we pause gas metering, and we're entering/inside
-    /// CREATE / CREATE2 frames. This is needed to make gas meter pausing work correctly when
-    /// paused and creating new contracts.
-    pub gas_metering_create: Option<Option<Gas>>,
+    /// If true then gas metering is paused.
+    pub pause_gas_metering: bool,
+
+    /// Stores paused gas per frame.
+    pub paused_frame_gas: HashMap<usize, Gas>,
+
+    /// Current execution frame, used for pause / resume gas metering.
+    pub current_frame: usize,
```

Comment:
This looks like it could just be Vec<> to push and pop from.

Reply:
yep, changed in https://github.com/foundry-rs/foundry/pull/8736/commits/8fb2f5b597ffca603114d2f50d7d726d21a0bb05